### PR TITLE
feat(planning_evaluator): add evaluation feature of trajectory lateral displacement

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
## Description

This PR introduces a new trajectory lateral displacement metric and enhances the stability evaluation capabilities of the planning evaluator by incorporating ego vehicle velocity and evaluation time parameters.

**The main purpose of this metric is to detect trajectory chattering (rapid oscillations in planned paths), which can lead to unstable vehicle behavior.**

- Added new metric `trajectory_lateral_displacement` to evaluate cumulative lateral displacement
- Added ego odometry support to the metrics calculator for velocity-aware evaluations
- Introduced `evaluation_time_s` parameter (default: 5.0s) for trajectory evaluation duration
- Added utility function `calc_lookahead_trajectory_distance` to calculate distance from ego position to trajectory end

### Implementation Details
- The new trajectory lateral displacement metric:
  - Evaluates the cumulative absolute lateral displacement between two trajectories
  - Uses ego vehicle velocity and evaluation time to determine the evaluation section length
  - Samples points along trajectories to measure lateral offsets
  - Considers trajectory length constraints based on ego position

## Related links

**Parent Issue:**

- Link

## How was this PR tested?

run simulator with foxglove visualization

https://github.com/user-attachments/assets/a7b2b8e6-3d50-4d01-b347-1f3ac97482c9

image

## Notes for reviewers

None.

#### Additions and removals

| Change type   | Parameter Name | Type     | Default Value | Description       |
|:--------------|:---------------|:---------|:--------------|:------------------|
| Added | `trajectory.evaluation_time_s`   | `double` | `5.0`         | time duration for trajectory evaluation in seconds |

## Effects on system behavior

None.

### CODE REVIEW

File: evaluator/autoware_planning_evaluator/src/metrics/stability_metrics.cpp [hunk 1]
```diff
@@ -96,5 +98,45 @@ Accumulator<double> calcLateralDistance(const Trajectory & traj1, const Trajecto
...
+  }
+
+  constexpr double num_evaluation_points = 10.0;
+  const double interval = evaluation_section_length / num_evaluation_points;
+
+  const auto resampled_traj1 = autoware::motion_utils::resampleTrajectory(traj1, interval);
+  double total_lateral_displacement = 0.0;
+  for (const auto & point : resampled_traj1.points) {
+    const auto p0 = autoware::universe_utils::getPoint(point);
+    const double dist = autoware::motion_utils::calcLateralOffset(traj2.points, p0);
```
Comment (maxime-clem) [inline evaluator/autoware_planning_evaluator/src/metrics/stability_metrics.cpp:134]: Don't you need to resample `traj2` too ?

File: evaluator/autoware_planning_evaluator/include/autoware/planning_evaluator/metrics/metrics_utils.hpp [hunk 2]
```diff
@@ -14,7 +14,6 @@
 
 #ifndef AUTOWARE__PLANNING_EVALUATOR__METRICS__METRICS_UTILS_HPP_
 #define AUTOWARE__PLANNING_EVALUATOR__METRICS__METRICS_UTILS_HPP_
-
```
Comment (satoshi-ota) [inline evaluator/autoware_planning_evaluator/include/autoware/planning_evaluator/metrics/metrics_utils.hpp:17]: remove diff

File: evaluator/autoware_planning_evaluator/src/metrics_calculator.cpp [hunk 3]
```diff
@@ -20,6 +20,7 @@
 #include "autoware/planning_evaluator/metrics/stability_metrics.hpp"
 #include "autoware/planning_evaluator/metrics/trajectory_metrics.hpp"
 #include "autoware/universe_utils/geometry/geometry.hpp"
+
```
Comment (satoshi-ota) [inline evaluator/autoware_planning_evaluator/src/metrics_calculator.cpp:23]: remove diff

File: evaluator/autoware_planning_evaluator/include/autoware/planning_evaluator/metrics/metrics_utils.hpp (hunk 2 above)
Comment (kyoichi-sugahara) [reply evaluator/autoware_planning_evaluator/include/autoware/planning_evaluator/metrics/metrics_utils.hpp]: adressed in 104fb0601b56d4666befa72ae147a171ec36e7f6

File: evaluator/autoware_planning_evaluator/src/metrics_calculator.cpp (hunk 3 above)
Comment (kyoichi-sugahara) [reply evaluator/autoware_planning_evaluator/src/metrics_calculator.cpp]: adressed in 2f06b233d799a6918f6c036ace480f7f87b07c76

File: evaluator/autoware_planning_evaluator/src/metrics/stability_metrics.cpp [hunk 4]
```diff
@@ -96,5 +98,45 @@ Accumulator<double> calcLateralDistance(const Trajectory & traj1, const Trajecto
...
+    total_lateral_displacement += std::abs(dist);
+  }
+  stat.add(total_lateral_displacement);
```
[the remark this answers is not in the document]
Comment (kyoichi-sugahara) [reply evaluator/autoware_planning_evaluator/src/metrics/stability_metrics.cpp]: @xtk8532704 
Nice catch thanks!
adressed in 051d7e8d4368f233f7a4e7b13271be2549bdca04 which includes changes unrelated to the comment though.

**Review by xtk8532704 — APPROVED** [review]
LGTM~

**Pull request merged**